src/CircleEditor/CircleEditorDocument.ts

```diff
@@ -191,4 +201,176 @@ export class CircleEditorDocument extends Disposable implements vscode.CustomDoc
+                tensor.quantization, Circle.QuantizationParametersT.prototype);
+          }
+          // sparsity parameters
+          if (tensor.sparsity) {
+            if (tensor.sparsity.dimMetadata) {
+              tensor.sparsity.dimMetadata =
+                  tensor.sparsity.dimMetadata.map((dimMeta: Circle.DimensionMetadataT) => {
+                    if (dimMeta.arraySegmentsType && dimMeta.arraySegments) {
+                      const sparseVectorClass = Object.entries(Types.SparseVector).find(element => {
+                        return dimMeta.arraySegmentsType === parseInt(element[0]);
+                      });
+                      if (sparseVectorClass && sparseVectorClass[1]) {
+                        dimMeta.arraySegments = Object.setPrototypeOf(
+                            dimMeta.arraySegments, sparseVectorClass[1].prototype);
+                      }
+                    } else {
+                      dimMeta.arraySegments = null;
+                    }
+                    if (dimMeta.arrayIndicesType && dimMeta.arrayIndices) {
+                      const sparseVectorClass = Object.entries(Types.SparseVector).find(element => {
+                        return dimMeta.arrayIndicesType === parseInt(element[0]);
+                      });
+                      if (sparseVectorClass && sparseVectorClass[1]) {
+                        dimMeta.arrayIndices = Object.setPrototypeOf(
+                            dimMeta.arrayIndices, sparseVectorClass[1].prototype);
+                      }
+                    } else {
+                      dimMeta.arrayIndices = null;
+                    }
+                    return Object.setPrototypeOf(dimMeta, Circle.DimensionMetadataT.prototype);
+                  });  // end map dimMeta
+
+              if (!tensor.sparsity.traversalOrder || !tensor.sparsity.traversalOrder.length) {
+                tensor.sparsity.traversalOrder = [];
+              }
+              if (!tensor.sparsity.blockMap || !tensor.sparsity.blockMap.length) {
+                tensor.sparsity.blockMap = [];
+              }
+              Object.setPrototypeOf(
+                  tensor.sparsity.dimMetadata, Circle.DimensionMetadataT.prototype);
+            }  // end if tensor.sparsity.dimMetadata
+
+            tensor.sparsity =
+                Object.setPrototypeOf(tensor.sparsity, Circle.SparsityParametersT.prototype);
+          }  // end if tensor.sparsity
+
+          return Object.setPrototypeOf(tensor, Circle.TensorT.prototype);
+        });
+
+        // operators
+        data.operators = data.operators.map((operator: Circle.OperatorT) => {
+          // case1 : custom operator
+          if (this._model.operatorCodes[operator.opcodeIndex].deprecatedBuiltinCode === 32) {
+            if (operator.builtinOptionsType || operator.builtinOptions) {
+              throw new Error;
+            }
+            // case2 : builtin operator
+          } else {
```

Comment:
Please do not make a distance between if and else

```suggestion
          } else {
          // case2 : builtin operator
```